packages/labs/react/src/create-component.ts

```diff
@@ -4,6 +4,47 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
+// Typecast that curries an Event type through a string to match
+// a prop name to a typed event callback.
+export type EventName<T extends Event = Event> = string & {
+  __event_type: T;
+};
+
+type Events = Record<string, EventName | string>;
```

Comment:
For our own understanding I'd like to comment this and `EventProps`. I keep forgetting why we have both.

I'd also like to consider another name for `Events`. Is it an `EventMap` maybe?

Reply:
How about `EventNameMap` since we also use `EventName` as that type?

Reply:
`EventName` -> `EventNames` -> `EventNameListeners` ?